test/performance/search/index_construction_benchmark.cpp

```diff
@@ -9,218 +9,188 @@
 
 #include <seqan3/alphabet/aminoacid/aa27.hpp>
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/core/type_traits/lazy.hpp>
 #include <seqan3/range/views/persist.hpp>
 #include <seqan3/range/views/rank_to.hpp>
-#include <seqan3/range/views/to.hpp>
-#include <seqan3/range/views/to_rank.hpp>
 #include <seqan3/search/fm_index/all.hpp>
 #include <seqan3/test/performance/sequence_generator.hpp>
 #include <seqan3/test/seqan2.hpp>
 
+
 #if SEQAN3_HAS_SEQAN2
 #include <seqan/index.h>
 #endif
 
-static void arguments(benchmark::internal::Benchmark* b)
+static constexpr int32_t max_length{50'000};
+static constexpr size_t seed{0x6126f};
+
+static void arguments(benchmark::internal::Benchmark * b)
 {
     for (int32_t length : {50, 5000, 50'000})
+    {
+        if (length > max_length)
+            throw std::logic_error{"Increase max_length to at least " + std::to_string(length)};
```

Comment:
```suggestion
            throw std::logic_error{"Increase max_length to at most " + std::to_string(max_length)};
```
:eyes: ?

Reply:
Mhm, either I decrease `length` or increase `max_length`; I only care about `length <= max_length` so I don't take a prefix longer than the sequence.